Merged in 2.9.1-alpha1 changes

jseyfert3 opened the pull request:
Forked probonopd/prusaslicer to jseyfert3/prusaslicer, cloned forked repo to PC, set upstream branch to prusa3d/prusaslicer, merged prusa3d/prusaslicer master branch into jseyfert3/prusaslicer build branch, pushed back to jseyfert3/prusaslicer.

That kicked off the build workflow, which completed successfully. I downloaded the AppImage, ran it, downloaded configs, sliced, and tried a feature that's new in the 2.9.1 alpha release. I then modified the hardcoded build name from 2.9.0-rc1 to 2.9.1-alpha1 in the GitHub workflow file, so once this PR is merged into probonopd/PrusaSlicer, the build that will kickoff will name the file 2.9.1-alpha1.

I did not fix the hardcoding of the release build, but given that Prusa will likely push out several more alpha/beta/rc releases before the 2.9.1 release, I will take a look and see if I can figure out how to not hardcode the release naming in the workflow file on a future update.

Jony01 committed:
﻿separate Voronoi graf and searching longest path into own files

Jony01 committed:
﻿Clean up voronoi test

Jony01 committed:
Node data constructor deosnt have own logic
Calculation of Parabola length by sum of line segments

Jony01 committed:
﻿Parabola length calculation.

Jony01 committed:
add Magic enum

Jony01 committed:
﻿add sample center circles

Jony01 committed:
Position of Support Point is connected with voronoi graph

Jony01 committed:
﻿convert Voronoi cell to polygon

Jony01 committed:
﻿Cell from over limit points

Jony01 committed:
﻿use geometry orientation

Jony01 committed:
﻿Align of SP

Jony01 committed:
﻿Separate draw function

Jony01 committed:
﻿Support island points are generated as unique ptr for option to move with them
+ add fix for cell 2 polygon transformation

Jony01 committed:
﻿Create ExPolygon from wide part of island

Jony01 committed:
Fix calculation of width on VD

Jony01 committed:
Successfull creation of field with indexes
Add visualization of Field

Jony01 committed:
Fix for sample between Tiny and Wide part of island

Jony01 committed:
﻿add minimal width to edge
Fix creation of Field and sampling outline

Jony01 committed:
Add uniform sampling of ExPolygon

Jony01 committed:
FIX> infinite loop because intersection is out of line

place on line mast be perpendicular to source line
but NOT to edge

Jony01 committed:
Fix. Now correct check when 2 vector have same direction

Jony01 committed:
﻿Compare of Sampling method

Jony01 committed:
﻿Center support point contain configuration

Jony01 committed:
﻿visualization inside of test with macro

Jony01 committed:
Fix data conversion warning

Jony01 committed:
Fix: include from parent directory (linux build)

Jony01 committed:
﻿Fix: Linux Warnings

Jony01 committed:
Fix: revert typedef

PrusaSlicer_With_Asan_Linux/248:
../src/libslic3r/SLA/SupportIslands/VectorUtils.hpp:76:25: error: need 'typename' before 'std::iterator_traits<_II>::value_type' because 'std::iterator_traits<_II>' is a dependent scope
../src/libslic3r/SLA/SupportIslands/VectorUtils.hpp:77:25: error: need 'typename' before 'std::iterator_traits<_Iter>::value_type' because 'std::iterator_traits<_Iter>' is a dependent scope
../src/libslic3r/SLA/SupportIslands/VectorUtils.hpp:78:24: error: need 'typename' before 'std::iterator_traits<_Iter>::difference_type' because 'std::iterator_traits<_Iter>' is a dependent scope
...

Jony01 committed:
Fix: data type for OsX

../src/libslic3r/SLA/SupportIslands/ParabolaUtils.cpp:38:48: error: no matching constructor for initialization of 'std::vector<Voronoi::Internal::point_type>' (aka 'vector<point_data<double> >')
    std::vector<Voronoi::Internal::point_type> parabola_samples(

Jony01 committed:
FIX: Slicer point now cast to boost point type by full constructor in parabola utils

Jony01 committed:
﻿remove magic enum

Jony01 committed:
Fix: Anotate unused variable to remove warnings

Jony01 committed:
FIX: remove warnings

../src/libslic3r/SLA/SupportIslands/PolygonUtils.cpp:12:10: warning: variable 'is_in_limits' set but not used [-Wunused-but-set-variable]
../src/libslic3r/SLA/SupportIslands/VoronoiGraphUtils.cpp:294:45: warning: unused variable 'success' [-Wunused-variable]

Jony01 committed:
FIX: Add convert Slic3r::Point to Voronoi::point_type

../src/libslic3r/SLA/SupportIslands/ParabolaUtils.cpp:92:22: error: no viable conversion from 'const Slic3r::Point' to 'VD::point_type' (aka 'point_data<double>')
    VD::point_type   source_point = parabola.focus;
                     ^              ~~~~~~~~~~~~~~
../src/libslic3r/SLA/SupportIslands/ParabolaUtils.cpp:39:33: error: no matching constructor for initialization of 'std::vector<VD::point_type>' (aka 'vector<point_data<double> >')
../src/libslic3r/SLA/SupportIslands/ParabolaUtils.cpp:85:33: error: no matching constructor for initialization of 'std::vector<VD::point_type>' (aka 'vector<point_data<double> >')

Jony01 committed:
Fix: Mac os build

../src/libslic3r/SLA/SupportIslands/ExpandNeighbor.cpp:23:63: error: 'value' is unavailable: introduced in macOS 10.14

Jony01 committed:
﻿Fix: change .value() connected with optional to dereferece by *

Os X do NOT support std::optional::value()
PrusaSlicer_OsX_Mojave:
../src/libslic3r/SLA/SupportIslands/ExpandNeighbor.cpp:23:50: error: 'value' is unavailable: introduced in macOS 10.14

Jony01 committed:
Align outline points

Jony01 committed:
Add allignable inner support island point

Jony01 committed:
﻿Add filter for end center line support instead of minimal distance.

Jony01 committed:
﻿Fix aligning
- function belong
- Calculation of move distance on outline

Separate utils tests

Jony01 committed:
FIX crop parallel line with y

Jony01 committed:
﻿Add outline angle rule to decide when support centre line

Jony01 committed:
﻿Fix build

Jony01 committed:
Fix configuration

Noisyfox committed:
SPE-2484: Port of Cura's multi-material interlocking from OrcaSlicer.

Co-authored-by: SoftFever <softfeverever@gmail.com>

hejllukas committed:
SPE-2387: Fix printing order of open perimeters with Arachne.

hejllukas committed:
SPE-2387: Order islands to minimize travels between them.

Jony01 committed:
Bug Fix

Jony01 committed:
Add comments

Jony01 committed:
Remove weird Z level for structure - points are mapped on surface of model.
It moves with support point by 3* layers ... 0.05*3 = 0.15 mm (invisible so mistake was ignored)
Head radius is 0.4 mm

Jony01 committed:
island centroid is not used soo I delete it

Jony01 committed:
Move minimal area into configuration to be able modify it from UI (in my opinion it is necessary)

Jony01 committed:
Remove tear preasure.

Jony01 committed:
Add new implementation of samling

Jony01 committed:
Supported point are created by distance(xy + z) to previous one

Jony01 committed:
﻿Autogenerate supportPoints by surface angle

Jony01 committed:
correction key to grid around zero

Jony01 committed:
Do not generate support tree when edit support points

Jony01 committed:
Sample only overhanging part of layer.

hejllukas committed:
SPE-2484: Fix int32 overflow inside multi-material interlocking.

Jony01 committed:
Distance of support point are modified by svg file.
 - Support points are stored in separate list(no need to collect from grid + no duplicit support points)
 - Grid contain only indices into result vector of support points
 - Each layer is made update of radiuses for existing support points
 - Each support point knows support radius for current processed layer
 - Layers knows z coordinate soo heights in preparation are copied into layer.

Jony01 committed:
squared distance for KD Tree can't be in CoordType when using coord_t as template parametre (because of overflow of value range)

Jony01 committed:
Change grid to KD tree for fast search of support points in max radius

Jony01 committed:
Remove points out of extended part.
Allow multiple sample of the same 2d part of object in multiple heights

Jony01 committed:
﻿Fixes of bad merges during rebase
Add missing includes into CMakeLists
IndexMesh change to AABBMesh
construct_voronoi become class function

Jony01 committed:
﻿Chenge is_last_neighbor to end_in_distance

Jony01 committed:
Partialy commented out SLA support point generator tests

Jony01 committed:
Fix of build

Jony01 committed:
Add default value for curve loaded from SVG file for fix test at this moment

Jony01 committed:
Fix sampling selfconnected thiny part of island sampled on voronoi skelet

Jony01 committed:
Use CGAL voronoi to calculate aligne of support point on the island

Jony01 committed:
Add crop line and fix crop ray

Jony01 committed:
Reduce points laying on the outline outline but still more often than field inside for initila sampling
Fix field endings
add check on inner

Jony01 committed:
Speed up support alignment

Jony01 committed:
Add support settings into GUI

Jony01 committed:
Fix tests

Jony01 committed:
Add option to store islands (some could make an issue this is tool to collect problematic islands)

Jony01 committed:
Fix:
/run/build/PrusaSlicer/src/libslic3r/SLA/SupportIslands/VectorUtils.hpp:34:13: error: ‘iota’ was not declared in this scope
   34 |         iota(idx.begin(), idx.end(), 0);
      |         ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~

Jony01 committed:
Advanced visualization to store island

Jony01 committed:
Fix for sampling and enhance visualization.

Jony01 committed:
Fix for island 40 from "Uniform sample test islands"
Last outline of Fieald is change back into first line

Jony01 committed:
Fix warnings:
⚠️ ../src/libslic3r/SLA/SupportIslands/SampleIslandUtils.cpp:1495:68: warning: lambda capture 'tiny_done' is not used [-Wunused-lambda-capture]
⚠️ ../src/libslic3r/SLA/SupportIslands/SampleIslandUtils.cpp:1830:23: warning: unused variable 'line' [-Wunused-variable]
⚠️ ../src/libslic3r/SLA/SupportIslands/SampleIslandUtils.cpp:624:6: warning: unused function 'is_points_in_distance' [-Wunused-function]
⚠️ ../src/libslic3r/SLA/SupportIslands/VoronoiDiagramCGAL.cpp:37:7: warning: unused function 'to_point_d' [-Wunused-function]
⚠️ ../src/libslic3r/SLA/SupportPointGenerator.cpp:359:6: warning: unused function 'exist_same_points' [-Wunused-function]
⚠️ ../src/slic3r/GUI/Gizmos/GLGizmoSlaSupports.cpp:813:27: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
⚠️ ../src/slic3r/GUI/Gizmos/GLGizmoSlaSupports.cpp:892:146: warning: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]

Jony01 committed:
Field sampling is idependent on translation and rotation of island

Jony01 committed:
Add separation of thin and thick part of island

hejllukas committed:
SPE-2501: Add an option to specify which extruder controls the bed temperature.

hejllukas committed:
SPE-2501: Improve the notification about the bed temperature difference.

Jony01 committed:
Searching for center thin part as start point for sampling

Jony01 committed:
Generate support points on thin part.

Need to fix some tips

Jony01 committed:
Add sampling of thick part of island with fixes found during debugging test cases.

Jony01 committed:
Clean up from unused code

Jony01 committed:
Rename to UniformSupportIsland

Jony01 committed:
Continue cleaning code
After thin&thic must be at least 2 support points

Jony01 committed:
Check for loop only when it is not start edge of thin part

Jony01 committed:
Fix duplicit point after alignment
+ fix detection if first_neighbor exist in process queue

Jony01 committed:
Comment changes which could lead to crash of tests in build enviroment

Jony01 committed:
Transfer Copy constructor of KDTreeIndirect into copy function
Reason: During Linux testing "fff_print_tests" it Fails from time to time.
ASAN version cause issue inside SeamPerimeters.hpp but I am not sure why.

Jony01 committed:
Modify GUI for searching support generator configuration
Rename config params.

Jony01 committed:
Add detection and creation of peninsula
with anotation of outline.

Jony01 committed:
Add support for peninsulas

Jony01 committed:
tbb paralelization for prepare part of support generation

Jony01 committed:
Store/Load support points 3mf
also store density change for object

Jony01 committed:
Iteractive tune of point density

kocikdav committed:
Missing error state in printer state array

kocikdav committed:
Sharing tokens among instances

Jony01 committed:
Testing version (commented out autogenerator parameters)
Edit configuration svg to change values to 160 percent of the density

Jony01 committed:
Separate propagation of permanent support points

kocikdav committed:
Mockup of UI notification when http communication performs retry.

Jony01 committed:
Add support for permanent supports on overhangs with propagation of influence into previous layers

Jony01 committed:
Fix KDTree search for closest points for coord_t
NOTE: Store squared distance into double

Jony01 committed:
Add support permanent points for island and peninsulas

kocikdav committed:
Multicast for login and logout.

Implemented on same base as single instance messages. 
Linux adds new Dbus object for this.
Message format for instance messaging changed to json.

Jony01 committed:
Add UI switch for show support structure

Jony01 committed:
Remove duplication of permanent points

Jony01 committed:
Density tooltip for whole slider
+ allowe value out of slider range
+ add minimal value
+ slider starts at 50%

Jony01 committed:
Remove "support_points_minimal_distance"

kocikdav committed:
debug logs

kocikdav committed:
use random number as pid on linux

Reason: Flatpak might use same pid for several instances.

Jony01 committed:
Remove permanent points out of mesh surface

kocikdav committed:
Perform reload on Connect tab when shown.

kocikdav committed:
Perform reload on Connect tab when shown.

Jony01 committed:
Do not remove permanent point influence in layers out of part.

Jony01 committed:
Move out prepare function for permanent point

Jony01 committed:
Accept density value after edit+enter

Jony01 committed:
add store/load 3mf flag,
that SLA support point is manualy edited by user
(NOTE: After user editat is lost informtion about supporting island)

Jony01 committed:
Density change quadraticaly supported radius

Jony01 committed:
SPE-2660: Add information message about possibly incompatible config,
amended by @lukasmatena

SachCZ committed:
SPE 2620: Update thumbnails on ObjectManipulation change.

kocikdav committed:
Do not use bool state behave_as_mater, instead do so only after token refresh event.

Jony01 committed:
Exist weird edge case, where expand function return empty result even when input is not empty.
Fix crash for files:
SPE-2518_SLA-Crash-3DBenchy.3mf
SPE-2518_SLA-Crash-RAB_2_pose_3.3mf
+ use expolygons as input for expanding layer part instead of polygons

Jony01 committed:
Add help for used colors of support points

Jony01 committed:
Show manual edited points with color during lock unique islands

Jony01 committed:
Add safety offset before search for overhangs
Prevent numerical instability => supports on the vertical wall

surynek committed:
Added simplified print head model (printer geometry) for the Original Prusa MK4S.

kocikdav committed:
Prevent renewal of access token into older token

kocikdav committed:
Pronibit after on show function when switching connect from notification

lukasmatena committed:
Added MK4S gantry model

lukasmatena committed:
Sequential test changed from hard error to soft error (allowing export)

surynek committed:
Added general bed zero position.

surynek committed:
Added method for returning bed printable volume and general rectangular bed when building PrinterGeometry.

surynek committed:
Modified object placement optimization function for general rectangular beds (not necessarily starting in [0,0] position).

lukasmatena committed:
Fixed crash in tests

lukasmatena committed:
Fixed encoding issue when loading json with gantries

surynek committed:
Modified Original Prusa XL geometry - increased hose and enclosure height

surynek committed:
Implemented non-rectangular (general convex) beds inside sequential scheduler.

surynek committed:
Removed assert check for non-rectangular bed (no longer needed).

surynek committed:
Fixed bug in polygon test within printable area - the test is going to be removed anyway.

surynek committed:
Removed polygon on bed test.

lukasmatena committed:
Pass non-rectangular beds, add undo/redo snapshot

kocikdav committed:
SPE-2663: Notification on  startup about secret repo not updating.

Warning dialog of failed update check.
Warning Dialog in Config wizard.
Icon in Wizard.
Rebase fix.

kocikdav committed:
SPE-2676: Use correct gcode name for each bed when uploading to Connect instead of the one from Connect.

Pass changed filename from Connect.

kocikdav committed:
SPE 2663: Missing clear in reset function.

Jony01 committed:
SPE-2671 fix
Change way of sampling inner outline of Field (thick part of island & peninsulas)
NOTE: Inner part(after offset border) could contain multiple ExPolygons and need to transfer from border information which line is outline

kocikdav committed:
New avatar path for download. Get without access token.

TODO: using webp as picture format.

kocikdav committed:
Changed title for BulkExportDialog

lukasmatena committed:
Ensure same order of objects when passing them to libseqarrange

Jony01 committed:
Fix for build test with new dependency

kocikdav committed:
SPE-2656: Fix of failed Load file in ScalableBitmap constructor.

enricoturri1966 committed:
SPE-2678 - Fixed rendering of tool marker model for sequential prints in preview

kocikdav committed:
SPE-2679: Missing error code in fs::exists causes crash.

kocikdav committed:
Missing include.

lukasmatena committed:
Seq arrange now handles selected-bed-only arrange

Jony01 committed:
Fix for SPE-2674 1/2
issue in file SPE-2674_mini13-armor-runner-v8.3mf

surynek committed:
Revisited simplified extruder model for MK4 and code cleaning.

lukasmatena committed:
Seq arrange now ignores unprintable volumes and maintains order of object which were not touched

lukasmatena committed:
Object labels in 3D scene show sequential order only for active bed

Jony01 committed:
SPE-2671_2 (detected by assert)
Solve last iteration

Jony01 committed:
Fix warnings
libslic3r/SLA/SupportPointGenerator.cpp:20:6: warning: unused function 'exist_point_in_distance' [-Wunused-function]
libslic3r/SLA/SupportPointGenerator.cpp:600:45: warning: lambda capture 'is_lower' is not used [-Wunused-lambda-capture]
libslic3r/SLA/SupportPointGenerator.cpp:728:15: warning: lambda capture 'sample_distance_in_um2' is not used [-Wunused-lambda-capture]
libslic3r/SLAPrintSteps.cpp:172:11: warning: lambda capture 'tree' is not used [-Wunused-lambda-capture]
slic3r/GUI/Gizmos/GLGizmoSlaSupports.cpp:790:40: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
slic3r/GUI/Gizmos/GLGizmoSlaSupports.cpp:90:12: warning: unused variable 'border' [-Wunused-variable]

hejllukas committed:
SPE-2501: Add support for blinking arrow for more items on the same tab.

YuSanka committed:
DoubleSliderForLayers: Added cog menu item "Sequential slider applied only to top layer"

surynek committed:
Revisited scheduling of object instances via object gluing, added object gluing across multple beds.

surynek committed:
Added object gluing across beds for scheduling of individual beds.

Jony01 committed:
FIX SPE-2674_symphysis.3mf 2/2
Island which can't create Voronoi Diagram are supported by one point in center of bounding box

surynek committed:
Bug fix in temporal scheduling constraint for object instances (object gluing).

lukasmatena committed:
Forbid seq arrange for a single bed in cases where it would reorder objects unexpectedly

hejllukas committed:
SPE-2484: Fix "Infill failed" during interlocking generation.

hejllukas committed:
SPE-2484: Restrict interlocking beam width parameter to meaningful values.

hejllukas committed:
Fix debug include in FillRectilinear.cpp.

kocikdav committed:
Missing Connect address resolving from service config.

lukasmatena committed:
Follow-up to 84de301:
- Fixed updating of model when printers are switched
- Fixed needless updates of model when seq printing is disabled (introduced in 8e5f5e1)

lukasmatena committed:
Exception handling for too large objects

lukasmatena committed:
Clean-up of tests

lukasmatena committed:
Turn off tests again, until they are fixed

surynek committed:
Added interface function for determining a pair of objects in sequential collision.

lukasmatena committed:
Propagate information about colliding objects into the UI

Jony01 committed:
remove Unused include from higher version of compiler

Jony01 committed:
Fix: Windows and Linux has different order of polygons after Clipper simplify
To unify behavior between platforms: Polygons are sorted on the end of function.

+ Remove Slic3r::ExPolygons simplify_polygons_ex(const Slic3r::Polygons &subject);  // different definition and declaration, unused function

kocikdav committed:
SPE 2663: Close notification on login.

lukasmatena committed:
Some UI polishing

lukasmatena committed:
Merge branch 'lm_seq_final'

lukasmatena committed:
Merge branch 'lh_extrusion_order' (#11658)

lukasmatena committed:
Remove the old warning dialog about FW version needed for bgcode

lukasmatena committed:
MK4-specific tweaks can now also be enabled by RAMMING_EXTRA in printer notes

lukasmatena committed:
Merge branch 'lh_different_materials'

lukasmatena committed:
Merge branch 'dk_avatar_new'

lukasmatena committed:
Bumped up version to 2.9.1-alpha0

hejllukas committed:
SPE-2484: Fix the "Input value is out of range" error repeatedly triggered because of lost precision during the conversion of double value to float.

kocikdav committed:
After rebase changes

kocikdav committed:
Missing declaration fix

kocikdav committed:
Custom set of unused characters in Bulk export.

rtyr committed:
Extruder clearance moved to printer settings.

lukasmatena committed:
Merge branch 'lh_orca_interlocking'

lukasmatena committed:
Merge branch 'dk_token_connect'

lukasmatena committed:
Merge branch 'dk_vendor_out'

lukasmatena committed:
Merge branch 'dk_connect_filename'

lukasmatena committed:
Update of taskbar icon on Win

surynek committed:
Reactivated polygon test module of libseqarrange.

kocikdav committed:
Changed log level for Connect message handling.

lukasmatena committed:
Do not use seq arrange when in SLA

lukasmatena committed:
Bumped up version to 2.9.1-alpha1

YuSanka committed:
PresetCollection: Fixed update_vendor_ptrs_after_copy().
Ptrs are updated for edited preset too.

Related to SPE-2690

surynek committed:
Revisited all test modules for sequential printing library.

lukasmatena committed:
Some fixes in phrases

Jony01 committed:
Fix for creating wide tiny change.
+ unify runtime by selecting position with smallest source index

hejllukas committed:
Fix int32 overflow.

Jony01 committed:
Change configuration based on test prints

One support: from(3.8mm) to(1.65mm) https://cfl.prusa3d.com/display/SLA/Single+Supporty
Two supports: from(8.3mm) to(6.5mm) https://cfl.prusa3d.com/display/SLA/Double+Supports+-+Rectangles
ThinThick hystereze: from(4.9 - 5.7mm) to (3.6 - 4.2mm) https://cfl.prusa3d.com/display/SLA/Double+Supports+-+Squares

Guesses
Thin Max Distance: from(9mm) to (5mm)
Thick inner max distance: from(9mm) to (6.5mm)
Thick outline max distance: from(6.75mm) to (4.875mm)
minimal_distance_from_outline - still touch edge of the island
maximal_distance_from_outline: from(3mm) to  (1.73mm)

lukasmatena committed:
Moved ProfilesSharingUtils files into CLI folder

YuSanka committed:
Follow-up 42d4f5c7: Reverted "export_sla" and "export_gcode" options into CLIActionsConfigDef

lukasmatena committed:
SPE-2691: Write 'prepared by' info from an env var, updated libbgcode

lukasmatena committed:
SPE-2691: Fixed resizing of thumbnails to maintain aspect ratio of the image

Jony01 committed:
Use support headDiameter from config
(generate support config when diameter is known)
+ overhangs sample curve is static inside of code(do not use svg config file more)
+ Move island configuration behinde macro

lukasmatena committed:
SPE-2687: Remove 1.5s pause after toolchange for XL

lukasmatena committed:
Merge branch 'lm_seq_final' (libseqarrange tests update)

phx-jonathan-seyfert committed:
Merging Prusa's latest master branch into my fork of probonopd/PrusaSlicer so I can test building an AppImage of the latest alpha release

jseyfert3 committed:
Update build_appimage.yml

Updated version number to 2.9.1-alpha1. Did not fix the hardcoded version number.

Jony01 committed:
Hide factory include into cpp file
+ remove unused minimal island area

Jony01 committed:
Do not generate data for support generator when supports are not needed

lukasmatena committed:
SPE-2696: Merge remote-tracking branch 'private/ys_cli_tweaks'

lukasmatena committed:
Changed one phrase and one logging level

probonopd commented:
At some point, it'd probably be less work for everyone to merge my changes into Prusa's repository?

surynek committed:
SPE-2702: Seq arrange data for HT90, printhead model, extended delta visualization

hejllukas committed:
SPE-2698: Fix crash during regions expansion.

kocikdav committed:
changed log level.

SachCZ committed:
SPE-2700 Scarf seam minor fixes

- Do not generate scarf seam on the first layer
- Fix links to the web manual pages

SachCZ committed:
SPE-2701: Disable scarf seam on thick bridges (#14190)

Jony01 committed:
SPE-2673
Support point generator ignor small part of model and do not support it

Jony01 committed:
Fix First swich between show/hide support structure and pad

Jony01 committed:
change icon button to text button

lukasmatena committed:
Merge branch 'lm_webslicer'

rtyr committed:
Sync with github.

lukasmatena committed:
Merge branch 'fs_SLA_supports'

surynek committed:
Bug fix in scheduling of object instances (for HT90)

lukasmatena committed:
Seq arrange better exception handling

surynek committed:
Corredted simplifier model (printer geometry) for HT90, a new final bounding box

YuSanka committed:
SLA: Fixed placement on the bed of the object after its auto Rotation
SPE-2750

YuSanka committed:
Localization: Fixed some phrases and updated POT
+ updated internal dictionaries
+ community dictionaries are merged with new POT

lukasmatena committed:
Fixed missing comma in json

lukasmatena committed:
Do not show HT90 rods when not printing sequentially

lukasmatena committed:
Merge branch 'lm_seq_ht90'

lukasmatena committed:
Bumped up version to 2.9.1-beta1

jseyfert3 committed:
Update build_appimage.yml

Update hardcoded version to 2.9.1-beta1

phx-jonathan-seyfert committed:
Merge remote-tracking branch 'upstream/master' into build

jseyfert3 commented:
Closing PR as I have updated my fork with the 2.9.1-beta1 changes now.
